ai-service/src/ai_service/embeddings/embedder.py

```
@@ -0,0 +1,138 @@
+
+    Returns:
+        Loaded SentenceTransformer model optimized for code embeddings.
+
+    Raises:
+        EmbeddingError: If model loading fails.
+    """
+    model_name = utils.get_env_var(constants.EMBEDDING_MODEL)
+    device = _get_device()
+
+    try:
+        logger.info(f"Loading embedding model: {model_name}")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)